src/boot/bootctl.c

```diff
@@ -519,7 +519,7 @@ static int copy_file_with_version_check(const char *from, const char *to, bool f
 static int mkdir_one(const char *prefix, const char *suffix) {
         char *p;
 
-        p = strjoina(prefix, "/", suffix);
+        p = path_join(prefix, suffix);
```

Comment:
this changes allocation from stack to heap. Needs a `_cleanup_free_` decorator on `char *p` hence.